hilti/runtime/src/types/stream.cc

```diff
@@ -113,6 +113,18 @@ int Chain::numberOfChunks() const {
     return n;
 }
 
+bool View::isComplete() const {
+    _ensureValid();
+
+    if ( _begin.isFrozen() )
```

Comment:
The only other use of `View::isFrozen` was in `MatchState::advance` which is now gone. Could you remove the global method and inline it here? That would help avoid having two pretty similar functions which could be easy to confuse.